### PR TITLE
CodeReview2 hickmadc (Daniel Hickman)

### PR DESCRIPTION
**Pull request by Razorwind1**
Great Project! I really love the idea and the UI mockups are noice :D

**Analysis**: Program could build and run (after my SDK Tools updates) and looked good in the emulator!
**Available**: Yep
**Documentation**: ReadMe is awesome (and what drew me to review the project in the first place).
**Comments**: I definitely prefer more explanatory in-code comments to cater to the less-knowledgeable coder, but it's preference to be sure.

Fun fact: I see you are using the same bottom_navigation_menu that my group is using! :)

(animation) Love the fun UI! The **_#1 UI addition_** I would recommend for a better user experience is: Make a **COOL ANIMAITON** for clicking the stars!! I will be much more likely to have fun with this app if there's a fun (perhaps sparkling?) animation with the star :D

(bug) You likely are working on it but, the list doesn't scroll down all the way.

(feature) There is no way to **_Un-Star_** an item -- a feature I would much desire if I were a user! (I don't see it in the requirements so just make sure that doesn't get over-looked :) )

### CODE REVIEW

File: app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@ import edu.uc.groupProject.topten.ui.main.PrivateListFragment
  * MainActivity class.
  */
 class MainActivity : AppCompatActivity() {
-    var mvm: MainViewModel = MainViewModel()
+
```
Comment (Razorwind1) [inline app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt:19]: never used...

File: app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt [hunk 2]
```diff
@@ -47,7 +47,7 @@ class MainActivity : AppCompatActivity() {
         */
 
 
-        this.supportActionBar?.hide();
+        this.supportActionBar?.hide()
```
Comment (Razorwind1) [inline app/src/main/java/edu/uc/groupProject/topten/MainActivity.kt:50]: Lol. Semi-colons again :)

File: app/src/main/java/edu/uc/groupProject/topten/service/FirestoreService.kt [hunk 3]
```diff
@@ -131,7 +131,7 @@ class FirestoreService {
                 if (document != null) {
                     val allListItems = ArrayList<ListItem>()
 
-                    var listItem: ListItem = ListItem(
+                    val listItem: ListItem = ListItem(
```
Comment (Razorwind1) [inline app/src/main/java/edu/uc/groupProject/topten/service/FirestoreService.kt:134]: It is practice is to use val until you decide you need to modify it (in which case you go back and define it as a var). "**Favoring Immutability**" is the principle here. There is always the chance a variable value will change unexpectedly, so this way you can be signaled when that is happening and can then decide if it's good functionality (and change it to a var) or if something is off. Here is a website that talks some about it along with other Kotlin Best Practices: https://batchofcode.com/2018/04/13/kotlin-best-practices.html#1-embrace-immutability

File: app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt [hunk 4]
```diff
@@ -19,9 +19,11 @@ data class TopTenList(
     var id:Int, var title:String, var description:String="", var isActive:Boolean,
     var category:String /* Possible category dto? */, var publishDate:Date,
      var expireDate: Date
+
```
Comment (Razorwind1) [inline app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt:22]: I like the comment you made here! Makes clear the thought-process you are going through and can spark discussion among the team.

File: app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt [hunk 5]
```diff
@@ -19,9 +19,11 @@ data class TopTenList(
...
 ){
     private var mlistItems:List<ListItem> = ArrayList<ListItem>()
 
+
```
Comment (Razorwind1) [inline app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt:26]: what does "m" stand for in mlistItems? Just curious (the IDE thinks it's a typo lol).

**Comment by Razorwind1** [comment]
Ok, now for the big thing (which is why 22 files are changed lol)

**_LAUNCHER ICON CHANGE_**
I saw you had a logo, which is super cool! So I thought, "why not make it into an app icon?" I just did this with my project (mostly following this video: https://www.youtube.com/watch?v=F_1I659RDCM ).

What I did for you in this project is the following:
1. Took the .png file you had and made it 512x512px (what launcher icons need to be)
2. Right-clicked on the "res" directory, selecting New --> Image Asset
3. I changed the "Foreground Layer" to the .png file, which has a transparent background, meaning the default background layer shows. So you can change the "Bakcground Layer" yourself to any color/gradient you create in a 512x512 image.

File: app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupProject/topten/dto/TopTenList.kt:26]: This is often a naming convention for private/internal variables.

File: app/src/main/java/edu/uc/groupProject/topten/service/FirestoreService.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/edu/uc/groupProject/topten/service/FirestoreService.kt:134]: Good idea.

**Comment by discospiff** [comment]
Several good suggestions here.  You can merge these, or simply make the changes manually, as they are fairly self contained.

**Pull request closed without merging**